src/test/java/com/hillel/items_exchange/controller/AuthControllerIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+package com.hillel.items_exchange.controller;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import com.github.database.rider.core.api.dataset.DataSet;
+import com.github.database.rider.spring.api.DBRider;
+import com.hillel.items_exchange.dto.UserRegistrationDto;
+import com.hillel.items_exchange.exception.BadRequestException;
+import com.hillel.items_exchange.exception.UnprocessableEntityException;
+import com.hillel.items_exchange.util.AuthControllerIntegrationTestUtil;
+import static com.hillel.items_exchange.util.JsonConverter.asJsonString;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import org.junit.jupiter.api.Test;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@DBRider
+@AutoConfigureMockMvc
+@Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:index-reset.sql")
+public class AuthControllerIntegrationTest extends AuthControllerIntegrationTestUtil {
+
+    private final MockMvc mockMvc;
+
+    @Autowired
+    public AuthControllerIntegrationTest(MockMvc mockMvc) {
+        super();
+        this.mockMvc = mockMvc;
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_shouldCreateNewUserAndReturnCreated() throws Exception {
```

Comment:
Add to test name word `Valid` for user and add validation of creation user from database side via Database-Rider.